### DISCUSSION TITLE
`fetch` having a different behavior on Vercel with input request

### DISCUSSION THREAD

**Discussion by asaroukou** [vercel/next.js#70047] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/asaroukou/fetch-request-vercel

### To Reproduce

Start the dev server & navigate to `/api?method=get` (to mock a get req) OR `/api?method=post` (to mock a post request).

### Current vs. Expected behavior

Current:
`/api?method=post` return text `Response from GET /api/foo`

Expected:
`/api?method=post` should return text `Response from POST /api/foo`

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.5.0
Binaries:
  Node: 22.8.0
  npm: 10.8.2
Relevant Packages:
  next: 14.2.8
  eslint-config-next: 13.4.20
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

### Additional context

When doing `fetch(new Request(), { method: "POST" })` I expect a post request to be done as stated by the fetch standard https://fetch.spec.whatwg.org/#requestinit, in `fetch(req, init)`, `init` should override `req` options

> If init["method"] [exists] then:
>    Let method be init["method"].

**Comment by samcx** [comment]
@asaroukou Thanks for flagging this! I was able to replicate the issue.

This is unlikely a <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> issue but a <picture data-single-emoji=":vercel:" title=":vercel:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/vercel/5e06be80d7237806.png" alt=":vercel:" align="absmiddle"></picture> one, but will confirm with the team.

**Comment by abhi12299** [comment]
<img width="604" alt="image" src="https://github.com/user-attachments/assets/3c6ab05c-4fa2-4a86-8c1b-290055070254">

Can confirm it is a vercel issue. Tried to reproduce it locally with `next dev` and `next start` but it only happens on vercel.

**Reply by smaeda-ks** [reply]
@asaroukou @abhi12299, thanks for flagging this. We've identified the issue in our serverless runtime and are working on releasing a fix. Will let you know once it's deployed globally. If this is super urgent, a quick dirty workaround would be to assign a random request header in your fetch call:

```js
fetch(new Request(), { method: "POST", headers: { "x-pseudo-header": "foo" }})
```

**Reply by smaeda-ks** [reply, marked as answer]
@asaroukou @abhi12299 The fix has now been rolled out to production. Any new deployments going forward should not see this issue. We're sorry for the trouble. We've added regression tests to make sure this won't happen again in the future.

**Comment by samcx** [comment]
@asaroukou Moving this out of issues in the meantime since it is not an issue with Next.js itself, but we are taking a look at this!

**Reply by asaroukou** [reply]
@samcx thanks for looking into it. We were using fetch this way to reuse a request and prevent code duplication. Specially when you hit the same endpoint on different route you don't want to duplicate your headers, etc. We can wait for a resolution; in the meantime we will work around the issue by NOT using an input `Request` instance.
